src/components/LMap.vue

```diff
@@ -131,6 +160,27 @@ watch(mapContainer, () => {
   disposeMap();
   createMapLayer();
 });
+
+/**
+ * Este watch trackea los markers si cambian y se actualizan si es así.
+ */
+watch([mapInstance, ():typeof props.markers => props.markers], () => {
+  disposeMarkers();
+  setMarkers();
+}, { deep: true }) ;
+
+/**
+ * Este watch trackea si el usuario ha dado permisos de ubicación y si los da resetea la vista
+ * a la ubicación del usuario
+ */
+watch(locationAccess, () => {
+  if (locationAccess.value === 'granted') {
+    createMapLayer();
+  } else if ((locationAccess.value === 'denied' || locationAccess.value === 'prompt') && !isNil(userMarker)) {
+    userMarker.remove();
+    userMarker = undefined;
+  }
+});
```

Comment:
```suggestion
}, { immediate: true });
```

https://vuejs.org/guide/essentials/watchers.html#eager-watchers

Por defecto, los watchers de Vue al registrarse no se ejecutan hasta que no suceda el primer cambio en el objeto (a partir de ahí ya también con todos los sucesivos). Esta opción hace que el callback se ejecute nada más se registre el watcher (es decir, también antes del primer cambio)

En nuestro caso, funciona porque si te fijas [en el código fuente del ``usePermission``](https://github.com/vueuse/vueuse/blob/main/packages/core/usePermission/index.ts#L86), el ``ref`` interno que maneja se inicializa en blanco (undefined) y luego se actualiza dentro de una Promesa, por lo que nuestro watcher lo coge y el componente aparece correctamente.

Pero esto podría no ser así en algún momento (imagínate que la librería VueUse actualiza el código de forma que el ref interno que ellos manejan no se inicialice vacío), entonces dejaría de funcionar. Para evitar regresiones (y como sabemos de antemano que este watcher se debe de ejecutar al inicializarse el componente), mejor marcarlo como inmediato.

Reply:
Lo añadido al watch de location pues